t/local/33_x509_create_cert.t

```diff
@@ -114,7 +107,22 @@ is(Net::SSLeay::X509_NAME_cmp($ca_issuer, $ca_subject), 0, "X509_NAME_cmp");
   like(my $crt_pem = Net::SSLeay::PEM_get_string_X509($x509), qr/-----BEGIN CERTIFICATE-----/, "PEM_get_string_X509");
   
   like(my $key_pem1 = Net::SSLeay::PEM_get_string_PrivateKey($pk), qr/-----BEGIN (RSA )?PRIVATE KEY-----/, "PEM_get_string_PrivateKey+nopasswd");        
-  like(my $key_pem2 = Net::SSLeay::PEM_get_string_PrivateKey($pk,"password"), qr/-----BEGIN (ENCRYPTED|RSA) PRIVATE KEY-----/, "PEM_get_string_PrivateKey+passwd");
+  SKIP: {
+      # Upcoming Net::SSLeay version 2.00 is likely to remove obsolete
+      # functionality. This includes updating PEM_get_stringPrivateKey
+      # default algorithm from EVP_des_cbc() to, for example,
+      # EVP_aes_128_cbc(). When this is done, this SKIP block and
+      # everything in it, besides the test itself, can be removed.
+      fail("Legacy provider still used with Net::SSLeay version $Net::SSLeay::VERSION") if $Net::SSLeay::VERSION =~ m/^2/s;
+      if (defined &Net::SSLeay::OSSL_PROVIDER_load &&
+	  !Net::SSLeay::OSSL_PROVIDER_load(undef, 'legacy'))
+      {
+	  my $des_warning = 'Warning: No legacy provider. DES in CBC mode used by PEM_get_string_PrivateKey by default is not available';
```

Comment:
Since Test::More prints this to the console, it might be better to shorten it to `No legacy provider` and mention the reason the legacy provider is needed for this test in the comment above.

Reply:
I can add more comments and shorten and update the `diag()` message to tell to check the comments in the text. Because it currently fails the test, the packager or end user is aware that something does not right when there's no default provider. My idea with a prominent message on the console is to try to ensure that the same person now still notices that something may not work even though the tests are passing.

The failure is sort-of expected behaviour but I want also to make sure that this is not hidden and later causing surprising failures.

Reply:
`diag()` is now shorter and tells what's it's for:
```
t/local/32_x509_get_cert_info.t ............. ok       
t/local/33_x509_create_cert.t ............... 1/141 # No legacy provider for PEM_get_string_PrivateKey
t/local/33_x509_create_cert.t ............... ok       
```